Source/AliveLibAO/Abe.cpp

```diff
@@ -2258,6 +2260,228 @@ __int16 Abe::TryMountElum_42E600()
+            || field_FC_current_motion == eAbeStates::State_68_LedgeHangWobble_428E50
+            || field_FC_current_motion == eAbeStates::State_65_LedgeDescend_428C00
+            || field_FC_current_motion == eAbeStates::State_67_ToOffScreenHoist_428C50)
+        {
+            shootKind = ShootKind::eHanging_1;
+        }
+
+        field_106_shot = 1;
+        field_100_health = FP_FromInteger(0);
+
+        switch (pBullet->field_10_type)
+        {
+            case BulletType::Type_0:
+            case BulletType::Type_1:
+            {
+                auto yPos = FP_FromInteger(24);
+                if (pBullet->field_20 > FP_FromInteger(0))
+                {
+                    yPos = -yPos;
+                }
+
+                auto pBlood = ao_new<Blood>();
+                if (pBlood)
+                {
+                    pBlood->ctor_4072B0(
+                        field_A8_xpos,
+                        pBullet->field_1C_ypos,
+                        yPos,
+                        FP_FromInteger(0),
+                        field_BC_sprite_scale,
+                        50);
+                }
+
+                if (shootKind != ShootKind::eEverythingElse_0)
+                {
+                    if (shootKind == ShootKind::eHanging_1)
+                    {
+                        field_FC_current_motion = eAbeStates::State_92_ForceDownFromHoist_4297C0;
+                        field_108_bMotionChanged = 1;
+                        field_106_shot = 0;
+                        field_114_gnFrame = 0;
+                    }
+                    else if (shootKind == ShootKind::eRolling_2)
+                    {
+                        if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX) == pBullet->field_20 > FP_FromInteger(0))
+                        {
+                            field_FE_next_state = eAbeStates::State_73_RollingKnockback_4291D0;
+                        }
+                        else
+                        {
+                            field_FE_next_state = eAbeStates::State_129_RollingKnockForward_4294F0;
+                        }
+                    }
+                }
+                else
+                {
+                    if (field_1A4_resources.res[0])
+                    {
+                        ToKnockback_422D90(1, 1);
+                    }
+                    else
+                    {
+                        ElumKnockForward_42E780(1);
+                    }
+                    if (field_10_anim.field_4_flags.Get(AnimFlags::eBit5_FlipX) != pBullet->field_20 > FP_FromInteger(0))
+                    {
+                        field_FC_current_motion = eAbeStates::State_128_KnockForward_429330;
+                    }
+                    field_108_bMotionChanged = 1;
+                    field_106_shot = 0;
+
+                    field_B4_velx = field_BC_sprite_scale * FP_FromDouble(7.8);
```

Comment:
can't this go in the else? seems strange to calc it and then overwrite it sometimes